### PR TITLE
Added development container

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: java-development/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM openjdk:8u252-slim-buster
```
Comment (cer) [inline java-development/Dockerfile:1]: Please use the same ordering as  https://github.com/eventuate-tram/eventuate-tram-examples-customers-and-orders/commit/ca1e1b4299da4c957c1e68f3f4ee4eebf088d94f - commonalities near the top, differences at the end.

File: docker-compose-java-development.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      - .:/eventuate-tram-sagas-examples-customers-and-orders-java-development
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://mysql/eventuate
+      SPRING_DATASOURCE_USERNAME: mysqluser
+      SPRING_DATASOURCE_PASSWORD: mysqlpw
+      SPRING_DATASOURCE_DRIVER_CLASS_NAME: com.mysql.jdbc.Driver
+      EVENTUATELOCAL_KAFKA_BOOTSTRAP_SERVERS: kafka:29092
+      EVENTUATELOCAL_ZOOKEEPER_CONNECTION_STRING: zookeeper:2181
+      SPRING_SLEUTH_ENABLED: "true"
```
Comment (cer) [inline docker-compose-java-development.yml:16]: java-development is not a service - it doesn't not need anything related to Sleuth!

File: docker-compose-java-development.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+version: '3.5'
+services:
+
+  java-development:
+    image: eventuateio/eventuate-tram-sagas-examples-customers-and-orders-java-development:latest
+    volumes:
+      - .:/eventuate-tram-sagas-examples-customers-and-orders-java-development
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://mysql/eventuate
```
Comment (cer) [inline docker-compose-java-development.yml:10]: Are any of these environment variables needed?

File: build.gradle [hunk 4]
```diff
@@ -55,4 +55,30 @@ dockerCompose {
...
+    }
 }
+
+task prepareJavaDevelopmentDockerImage(type:Exec) {
+    commandLine './prepare-java-development-docker-image.sh'
+}
+
+javaDevelopmentImageComposeBuild.dependsOn(prepareJavaDevelopmentDockerImage)
+
+task buildAndStartServicesMySql(type: GradleBuild) {
```
Comment (cer) [inline build.gradle:76]: This name is not consistent with `eventuate-tram-examples-customers-and-orders`, which uses `buildAndStartServicesMySqlBinlog` - I want to have consistent instructions for all projects. `Binlog` suffix can be removed.

File: prepare-java-development-docker-image.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#! /bin/bash -e
+
+TARGET=java-development/build
+rm -fr ${TARGET}
+mkdir ${TARGET}
+
+#tar -cf - *gradle* */src/* */build.gradle | tar -C ${TARGET} -xf -
```
Comment (cer) [inline prepare-java-development-docker-image.sh:7]: remove comments like this

File: publish-docker-images.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#! /bin/bash -e
+
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+
+export DOCKER_IMAGE_TAG=latest
+
+docker login -u ${DOCKER_USER_ID?} -p ${DOCKER_PASSWORD?}
+
+./gradlew assemble mysqlComposeBuild mysqlComposePush
```
Comment (cer) [inline publish-docker-images.sh:9]: Why is mysql being pushed?

Comment (dartartem) [reply publish-docker-images.sh:9]: Chris, I cannot find where did I get it, I will remove it

File: prepare-java-development-docker-image.sh (hunk 5 above)
Comment (dartartem) [reply prepare-java-development-docker-image.sh:7]: ok

File: java-development/Dockerfile (hunk 1 above)
Comment (dartartem) [reply java-development/Dockerfile:1]: ok

File: docker-compose-java-development.yml (hunk 2 above)
Comment (dartartem) [reply docker-compose-java-development.yml:16]: ok

File: docker-compose-java-development.yml (hunk 3 above)
Comment (dartartem) [reply docker-compose-java-development.yml:10]: It seems no

File: build.gradle (hunk 4 above)
Comment (dartartem) [reply build.gradle:76]: ok, I will remove it from eventuate-tram-examples-customers-and-orders

**Pull request closed without merging**